src/couch_dist/README.md

```diff
@@ -0,0 +1,160 @@
+# couch_dist
+
+`couch_dist` implements a custom distribution protocol -- `couch`, which allows
+nodes to connect using different protocols, such as Transport Layer Security
+(TLS) and Transmission Control Protocol (TCP).
+
+`TLS` can provide extra verification and security, but requires proper
+certificates and configuration to set up the environment.
+
+## Set up a custom Erlang distribution
+
+1. Specify the distribution protocol in `vm.args`
+2. Specify different distribution protocols for different nodes in `vm.args`
+3. Generate certificates using `certs`
+4. Specify security and other SSL options in `couch_dist.conf`
+
+Examples:
+
+1. `vm.args`:
+
+      ```vm.args
+      -proto_dist couch
+      -couch_dist no_tls '"clouseau@127.0.0.1"'
+      -ssl_dist_optfile </absolute/path/to/couch_dist.conf>
+      ```
+
+2. `couch_dist.conf`:
+
+    - `erlserver.pem`: contains the certificate and its private key.
+    - `{verify, verify_peer}`: you can specify the hostname with `{server_name_indication, <hostname>}`.
+    - `{fail_if_no_peer_cert, true}`: should be used on the server side only,
```

Comment:
`fail_if_no_peer_cert` should be set on client and server.

Reply:
The connection succeeds on OTP 24 and 25 but fails on OTP 26.
`./dev/remsh-tls` fails to connect to node1 if `{fail_if_no_peer_cert, true}` is present on the client side.

See https://www.erlang.org/blog/otp-26-highlights/#ssl-improved-checking-of-options

Reply:
How about adding this to the `gen_cert` script?
Generate different `couch_dist.conf` files for different erlang versions.
```bash
ERL_VER=$(erl -eval "io:put_chars(erlang:system_info(otp_release)), halt()." -noshell)
if [ $ERL_VER -ge "26" ]; then
  OPTS="{verify, verify_peer}"
else
  OPTS="{verify, verify_peer},
  {fail_if_no_peer_cert, true}"
fi
```

Reply:
> In OTP 26, the checking of options is strengthened to return errors for incorrect options that used to be silently ignored. For example, ssl now rejects the fail_if_no_peer_cert option if used for the client:

> In OTP 25, the option would be silently ignored.

https://www.erlang.org/blog/otp-26-highlights/#ssl-improved-checking-of-options

Or just leave `{fail_if_no_peer_cert, true}` on the server side. Because when we specify `{verify, verify_peer}`, it requires us to provide cacertfile, certfile, and keyfile to connect successfully.

Reply:
Revert `gen_cert` script back, so `{fail_if_no_peer_cert, true}` would only appear on server config side.
Also removed `couch_dist` from `couch.app.src` file, and tested it with OTP 24/25/26, worked on my end.